### DISCUSSION TITLE
ImportError: no module named 'pybricks.pupdevice'

### DISCUSSION THREAD

**Discussion by Ahardor** [pybricks/support#715] [Q&A]
Trying to run code below

# ColorSensor Example
    #!/usr/bin/env pybricks-micropython
    
    from pybricks.parameters import Port
    from pybricks.pupdevices import ColorDistanceSensor
    from pybricks.tools import wait
    
    #Initialize the sensor.
    sensor = ColorDistanceSensor(Port.A)
    
    while True:
        # Read the color.
        color = sensor.color()
    
        # Print the measured color.
        print(color)
    
        # Move the sensor around and see how
        # well you can detect colors.
    
        # Wait so we can read the value.
        wait(100)

And getting this log:

    Starting: brickrun --directory="/home/robot/fff" "/home/robot/fff/beep.py"
    ----------
    Traceback (most recent call last):
      File "/home/robot/fff/beep.py", line 4, in <module>
    ImportError: no module named 'pybricks.pupdevice'
    ----------
    Exited with error code 1.

Other listed modules don't cause any errors, only 'pybricks.pupdevices'

**Comment by dlech** [comment]
EV3 doesn't support Powered Up devices. Which sensor are you trying to use?

**Reply by Ahardor** [reply]
Mindstorms color detections sensor

**Reply by dlech** [reply]
This one? https://pybricks.com/ev3-micropython/ev3devices.html#color-sensor

**Reply by dlech** [reply]
It would be `from pybricks.ev3devices import ColorSensor`.

**Reply by Ahardor** [reply]
Thank you very much
